foam_controlwrapper/foam_mesh.py

```diff
@@ -657,9 +837,104 @@ def generate_uuidmapping(self, nPoints, nEdges, nFaces, nCells):
             self._foamCellLabelToUuid[cell] = uuid
 
     def _generate_uuid(self):
-        """Provides an uuid for the object
+        """ Provides an uuid for the object
 
         Provides san uuid as defined in the standard RFC 4122
         """
 
         return uuid.uuid4()
+
+    def modifyNumerics(self, SP):
```

Comment:
API suggestion: The ``run``, ``modifyNumerics`` and/or ``write_foammesh`` methods currently in FoamMesh could be defined outside the modeling engine class as module level utility functions.  Some changes might be required.